### PR TITLE
HDDS-5682: EC: Provide replication config option from CLI when creating bucket.

### PR DESCRIPTION
**Pull request by umamaheswararao**
## What changes were proposed in this pull request?

When creating bucket we should provide an option to pass the replication details. SO, that they can be persisted as bucket default replication.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5682

## How was this patch tested?

Added tests to verify it.

### CODE REVIEW

**Comment by sodonnel** [comment]
Summarising a conversation we had on slack.

I'd prefer to avoid creating a new mini-cluster to test this, as each time we add a new one, it adds about 40 seconds to the build time.

For the CLI logic, testing it handles its arguments properly and constructs the correct call to the Client API, a unit test similar to TestDecommissionSubCommand is ideal, with a mocked client.

As the test here was taken from TestOzoneShellHA, it already creates a mini-cluster in a BeforeClass method, and runs many tests, so we could also runs these tests there but extending that test too.

**Comment by umamaheswararao** [comment]
Good suggestion @sodonnel. It make sense to me to move the tests to TestOzoneShellHA, to avoid additional mini cluster startup. I did verify end-end behavior in the tests, example, create bucket with EC, and create key which should be in EC. Means it is depending on om state. SO, for this instance I would go with TestOzoneShellHA as I can verify that behavior.  
I will post the revised patch shortly.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java [hunk 1]
```diff
@@ -60,6 +66,17 @@
       defaultValue = "LEGACY")
   private AllowedBucketLayouts allowedBucketLayout;
 
+  @Option(names = {"--replication", "-r"},
+      description = "Replication value. Example: 3 (for Ratis type) or 1 ( for"
+          + " standalone type). Inthe case of EC type, the following valid"
```
Comment (sodonnel) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java:71]: Minor typo "Inthe" -> "in the".

Also, I wonder we should avoid being so specific about the EC types allowed, as that may change over time, and then we will forget to change this help text. Perhaps say "In the case of EC, pass DATA-PARITY, eg 3-2, 6-3, 10-4".

Comment (umamaheswararao) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java:71]: Thanks for the review again. It make sense, I will fix the description shortly.

**Review by sodonnel — APPROVED** [review]
+1, LGTM

**Pull request merged**